src/plugins/intel_gpu/src/plugin/ops/parameter.cpp

```diff
@@ -24,7 +24,7 @@ namespace ov::intel_gpu {
 static void CreateParameterOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v0::Parameter>& op) {
     auto input_pshape = op->get_partial_shape();
     if (!p.use_new_shape_infer() && input_pshape.size() < 4) {
-        input_pshape.insert(input_pshape.end(), 4 - input_pshape.size(), ov::Dimension(1));
+        input_pshape.insert(input_pshape.begin(), 4 - input_pshape.size(), ov::Dimension(1));
```

Comment:
I think this might affect many models where inputs are expected to follow the [B, F, Y, X] order, with the possibility that the F, Y, and X dimensions may be omitted. In that case, this PR would change the input to [1, 1, 1, B], which wouldn't be expected
I'd suggest applying more matmul-specific change. Both the gemm and fully_connected primitives have input_rank, weight_rank, and input_size, weights_rank parameters respectively, which can be used to restore the original shape rank for legacy shape inference